src/ServiceControlInstaller.Engine/Database/DatabaseMigrations.cs

```diff
@@ -67,6 +67,8 @@ public static void RunDataMigration(Action<string> updateProgress, string instal
                     }
                 }
             } while (attempts < 2);
+
+            File.Create(Path.Combine(logPath, $"{DateTime.UtcNow.ToFileTimeUtc().ToString()}.upgrade")).Close();
```

Comment:
Would it make sense to create an Upgrade Marker File service or something to hide/encapsulate the implementation details of the marker file?

Reply:
How would that be different to this file?

Reply:
@mikeminutillo I gave it a try over here https://github.com/Particular/ServiceControl/commit/c68b2d44ff00ee03d2214bcaf41168610dca370d but then I realized that the engine does not reference anything from SC. If we want to go on this path we have to link the source file in the engine. Is it worth it?